### PR TITLE
Fix #6927 1:1 NAT validate address family

### PR DESCRIPTION
**Pull request by phil-davis**
Ensure that all the manually-entered addresses come from the same
address family - i.e. they are all either IPv4 or IPv6 addresses.

### CODE REVIEW

File: src/usr/local/www/firewall_nat_1to1_edit.php [hunk 1]
```diff
@@ -175,8 +186,21 @@
 
 	/* For src, user can enter only ip's or networks */
 	if (!is_specialnet($_POST['srctype'])) {
-		if (($_POST['src'] && !is_ipaddr($_POST['src']))) {
-			$input_errors[] = sprintf(gettext("%s is not a valid internal IP address."), $_POST['src']);
+		if ($_POST['src']) {
+			$srcipaddrtype = is_ipaddr($_POST['src']);
+			if ($srcipaddrtype) {
+				if ($extipaddrtype) {
+					if ($srcipaddrtype != $extipaddrtype) {
```
Comment (rbgarga) [inline src/usr/local/www/firewall_nat_1to1_edit.php]: Both if's can be grouped as
```
if ($extipaddrtype && ($srcipaddrtype != $extipaddrtype)) {
```
and reduce indent level

File: src/usr/local/www/firewall_nat_1to1_edit.php [hunk 2]
```diff
@@ -186,8 +210,36 @@
...
 	/* For dst, user can enter ip's, networks or aliases */
 	if (!is_specialnet($_POST['dsttype'])) {
-		if (($_POST['dst'] && !is_ipaddroralias($_POST['dst']))) {
-			$input_errors[] = sprintf(gettext("%s is not a valid destination IP address or alias."), $_POST['dst']);
+		if ($_POST['dst']) {
+			if (is_ipaddroralias($_POST['dst'])) {
+				$dstipaddrtype = is_ipaddr($_POST['dst']);
+				if ($dstipaddrtype) {
+					if ($extipaddrtype) {
+						if ($dstipaddrtype != $extipaddrtype) {
```
Comment (rbgarga) [inline src/usr/local/www/firewall_nat_1to1_edit.php]: Same here, if's can be grouped

File: src/usr/local/www/firewall_nat_1to1_edit.php [hunk 3]
```diff
@@ -186,8 +210,36 @@
...
+							$input_errors[] = sprintf(
+								gettext("The external IP address (%s) and destination IP address (%s) are of different address families.") .
+									get_must_be_both_text(),
+								$_POST['external'],
+								$_POST['dst']);
+						}
+					}
+					if ($srcipaddrtype) {
+						if ($dstipaddrtype != $srcipaddrtype) {
```
Comment (rbgarga) [inline src/usr/local/www/firewall_nat_1to1_edit.php]: Same here, both if's can be grouped

**Comment by phil-davis** [comment]
Good thought, "if" statements condensed as suggested.

**Comment by rbgarga** [comment]
@phil-davis can you please rebase or resolve conflicts online? It's conflicting and cannot be merged

**Comment by phil-davis** [comment]
See #3293

**Pull request closed without merging**